src/functions.jl

```diff
@@ -191,34 +191,44 @@ for T in (:Taylor1, :TaylorN)
                 @inbounds c[0] = exp(constant_term(a))
                 return nothing
             end
-
-            @inbounds for i = 0:k-1
+            if $T == Taylor1
+                @inbounds c[k] = k * a[k] * c[0]
+            else
+                @inbounds mul!(c[k], k * a[k], c[0])
+            end
+            @inbounds for i = 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * a[k-i] * c[i]
                 else
                     mul!(c[k], (k-i) * a[k-i], c[i])
                 end
             end
             @inbounds c[k] = c[k] / k
-
             return nothing
         end
 
         @inline function log!(c::$T{T}, a::$T{T}, k::Int) where {T}
             if k == 0
                 @inbounds c[0] = log(constant_term(a))
                 return nothing
+            elseif k == 1
+                @inbounds c[1] = a[1] / constant_term(a)
+                return nothing
```

Comment:
I think this is not really needed (the same result is obtained from the rest of the code below), though it may actually speed things up a little. Have you done some benchmarks on its usefulness?

Reply:
I'll do some benchmarks and report back!

Reply:
I just did some benchmarks and performance looks essentially the same for `Taylor1{Float64}`. On the other hand, for `Taylor1{Taylor1{Float64}}` performance is a bit better with the `elseif k == 1 ...` block

Reply:
Thanks for reporting. Using nested Taylor1's is a good test for benchmarking. Then, I think it is worth leaving it there.